### ISSUE TITLE
Incorrect parsing of strings that contain non-delimiter comma

### ISSUE THREAD

**Issue by bpepple** [Kareadita/Kavita#3026]
### What happened?

For example, let's say the _comicinfo.xml_ of a comic has a list of teams, which is a comma delimited field, and one of the teams is `Infinity, Inc`. Currently Kavita imports it as `Infinity` & `Inc`, instead of as `Infinity, Inc`

### What did you expect?

When importing the metadata any string (whether story arc, teams, characters, etc) that contain a comma in it's name is not split in error.

Provided it's metadata was written [correctly](https://en.wikipedia.org/wiki/Comma-separated_values#General_functionality), that string should be be be double quoted so when it's read by Kavita or other software the data can be extracted correctly.

### Kavita Version Number - If you don not see your version number listed, please update Kavita and see if your issue still persists.

0.8.1 - Stable

### What operating system is Kavita being hosted from?

Docker (Dockerhub Container)

### If the issue is being seen on Desktop, what OS are you running where you see the issue?

Linux

### If the issue is being seen in the UI, what browsers are you seeing the problem on?

Firefox

### If the issue is being seen on Mobile, what OS are you running where you see the issue?

Android

### If the issue is being seen on the Mobile UI, what browsers are you seeing the problem on?

Firefox

**Comment by majora2007** [comment]
Hmm, I'm not sure how to handle this. Code isn't going to know that Joe, Joe, A, JoeB is 3 strings of "Joe", "Joe, A", and "JoeB". I'll reach out to see what the spec says about handling this.

**Comment by bpepple** [comment]
The Anansi Project doesn't cover it as far as I'm aware (and really using a comma as a delimiter on list of names isn't a great idea), but here's how I handled it for my python projects that write the `comicinfo.xml`, which might provide a bit of help: https://github.com/Metron-Project/darkseid/blob/222c434e279dcd0ea5460eb4c301a5463490e98a/darkseid/utils.py#L33

Using a regex that looks for items in quotes and items between commas like this should work:
```
(?:^|,)(\"(?:[^\"])*\"|[^,]*)
```

**Comment by majora2007** [comment]
I agree that it's not a good idea, but the spec is the most supported, so my hands are a bit tied. Quoting the string could work and Kavita could identify that pattern and de-quote. 

I'll take a look in the next stable window as the upcoming stable is days away.

**Comment by bpepple** [comment]
Doesn't Comicrack just ignore the the quotes, so it would parse it as `"Infinity` & `Inc."` instead of `Infinity` & `Inc.`, which are both incorrect.

I don't quite understand handcuffing yourself to project that's been dead for years and caring over their bugs. Anyway, just my 2¢.

**Comment by majora2007** [comment]
I'm not sure, I've never used Comicrack before. When I started Kavita I didn't know anything about the metadata scene. As of now, I just want users that have already tagged their collection to have their metadata usable. 

If I just change to support some new standard, it will be useless unless the major taggers and players in the scene support it. Kavita currently supports ComicInfo and some non-breaking enhancements (since I couldn't get Anansai project to approve new tags).

**Comment by bpepple** [comment]
I guess my thinking this **isn't** a breaking change, since the clients can still parse them, which is how I identified this bug in the first place. Regardless, this is obviously your decision on how you want to handle it.

**Comment by majora2007** [comment]
Right, I'm not implying it is. All I'm saying is that my next stable release (v0.8.2) is right around the corner, so I will look into this after that. I think you might have misunderstood that I was talking about a spec breaking change.

**Comment by bpepple** [comment]
> Right, I'm not implying it is. All I'm saying is that my next stable release (v0.8.2) is right around the corner, so I will look into this after that. I think you might have misunderstood that I was talking about a spec breaking change.

Yeah, looks like I misunderstood what you were saying. Sorry.

**Comment by majora2007** [comment]
I'll be closing this out. I'm stuck with the ComicInfo standard until I add MetronInfo support.
